connections/ethereum/connection.go

```diff
@@ -165,6 +165,34 @@ func (c *Connection) SafeEstimateGas(ctx context.Context) (*big.Int, error) {
 	}
 }
 
+func (c *Connection) SafeEstimateGasTipCap(ctx context.Context, baseFee *big.Int) (*big.Int, *big.Int, error) {
+	if c.maxGasPrice.Cmp(baseFee) < 0 {
+		return nil, nil, fmt.Errorf("maxGasPrice (%v) < BaseFee (%v)", c.maxGasPrice, baseFee)
+	}
+
+	var tip *big.Int
+	tip, err := c.conn.SuggestGasTipCap(context.TODO())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	gasFeeCap := new(big.Int).Add(
+		tip,
+		new(big.Int).Mul(baseFee, big.NewInt(2)),
```

Comment:
where did this 2 cames from?

Reply:
I saw this is how geth suggested no specified `gasFeeCap` (https://github.com/ethereum/go-ethereum/blob/28ba686cbfa6a46a9b4f03b620460867d502f34e/accounts/abi/bind/base.go#L270). I kept the logic as  paying a little higher than the suggested tip + baseFee seemed like a good way to guarantee tx inclusion, especially in the case of a gas spike.

Reply:
1 or 2 is the suggest amount as a default in the beginning until we see how it plays out. This was discussed on a call, I'll see if i can find the cliff notes